pulp_smash/tests/platform/cli/test_content_sources.py

```diff
@@ -0,0 +1,97 @@
+# coding=utf-8
+"""Tests for list/refresh/delete content sources."""
+from __future__ import unicode_literals
+
+import unittest2
+from packaging.version import Version
+
+from pulp_smash import cli, config
+from pulp_smash.constants import CONTENT_SOURCE_ID
+
+
+def _get_content_source_ids(server_config):
+    """Get the id list of all content sources, or None if the list is empty.
+
+    :param server_config: pulp_smash.config.ServerConfig server_config:
+        Information about the Pulp server being targeted.
+    :return: None if the list of content sources is empty, or the IDs of all
+        content sources as ``list``.
+    """
+    keyword = 'Source Id:'
+    completed_proc = cli.Client(server_config).run(
+        'pulp-admin content sources list'.split()
+    )
+    lines = [
+        line for line in completed_proc.stdout.splitlines()
+        if keyword in line
+    ]
+    assert len(lines) >= 0
+    if len(lines) == 0:
+        return None
+    return [line.split(keyword)[1].strip() for line in lines]
+
+
+class RefreshAndDeleteContentSourcesTestCase(unittest2.TestCase):
+    """Test whether pulp-admin client can refresh and delete content source.
+
+    This test case targets Pulp #1692`_ and `Pulp Smash #141`_. The
+    `content sources`_ documentation describes the CLI syntax. The
+    test steps are as follows:
+
+    1. Create configuration file of a content source.
+    2. Check whether the content sources list is empty.
+    3. Refresh all content sources. Verify no errors are reported.
+    4. Refresh a specified content source. Verify no errors are reported.
+    5. Remove the specified content source. Verify that the source is
+        actually deleted.
+
+    .. _Pulp #1692:  https://pulp.plan.io/issues/1692
+    .. _Pulp Smash #141: https://github.com/PulpQE/pulp-smash/issues/141
+    .. _content sources:
+        http://pulp.readthedocs.io/en/latest/user-guide/content-sources.html
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Verify if content resources exist."""
+        cls.cfg = config.get_config()
+        if cls.cfg.version < Version('2.9'):
+            raise unittest2.SkipTest('This test requires Pulp 2.9 or greater.')
+        cls.client = cli.Client(cls.cfg)
+        cls.source_id = CONTENT_SOURCE_ID
+        cls.source_ids = _get_content_source_ids(cls.cfg)
+        if cls.source_ids is None:
+            raise unittest2.SkipTest(
+                'This test requires a content source defined '
+                'in `/etc/pulp/content/sources/conf.d`'
```

Comment:
Actually the content source was manually added into folder. As soon as there's a `content.conf` in the folder `/etc/pulp/content/sources/conf.d/`, Pulp could automatically read in that file...

Also, from the docs, they assume an existing content source in Pulp...and they provide no pulp-admin CLI tool of adding it...